src/Semaphore/SemaphoreCompletion.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';

```

Comment:
Is it possible to wrap this with the checkbox?  It could simplify the CSS positioning and remove some of the boilerplate logic.

Reply:
I would just update the checkbox control with a "showPotential" property and put both elements in a DIV:

```html
<div>
    <input … /><span>{potential}</span>
</div>
```

Then in code insert `<space>A` for potential